lib/codecs/src/decoding/framing/chunked_gelf.rs

```diff
@@ -128,7 +165,76 @@ impl MessageState {
     }
 }
 
-#[derive(Debug, Snafu, PartialEq, Eq)]
+#[derive(Debug, PartialEq, Eq)]
+pub enum ChunkedGelfDecompression {
+    Gzip,
+    Zlib,
+    None,
+}
+
+impl ChunkedGelfDecompression {
+    // TODO: add tests for compression detection
+    pub fn from_magic(data: &Bytes) -> Self {
+        if data.starts_with(GZIP_MAGIC) {
+            debug!("Detected Gzip compression");
+            return Self::Gzip;
+        }
+
+        if data.starts_with(ZLIB_MAGIC) {
+            if let Some([first_byte, second_byte]) = data.get(0..2) {
```

Comment:
inspired from https://github.com/grafana/go-gelf/blob/25db8704bcf3f484c958312cd0cc49e5c768dcf1/gelf/reader.go#L120 and https://github.com/Graylog2/graylog2-server/blob/1bdeca5f8ad3d31ff0c1c7981978ae8c4d300ebc/graylog2-server/src/main/java/org/graylog2/inputs/codecs/gelf/GELFMessage.java#L157. This is based on [Zlib RFC](https://datatracker.ietf.org/doc/html/rfc1950)